src/managers/PermissionOverwriteManager.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const BaseManager = require('./BaseManager');
+const { TypeError } = require('../errors');
+const PermissionOverwrites = require('../structures/PermissionOverwrites');
+const Role = require('../structures/Role');
+const Collection = require('../util/Collection');
+const { OverwriteTypes } = require('../util/Constants');
+
+/**
+ * Manages API methods for guild channel permission overwrites and stores their cache.
+ * @extends {BaseManager}
+ */
+class PermissionOverwriteManager extends BaseManager {
```

Comment:
`PermissionOverwriteManager#set`, `edit`, `create` and `delete` should return `this` to allow chaining.

Reply:
It's not consistent; and chaining can still be done anyway, so I don't really think so that'd be necessary.

Reply:
When these methods were defined in `GuildChannel`, it was relevant to return the `this`, to allow chaining. But since they are now defined in the manager, it don't make sense to return the `GuildChannel` object. At least they should return `void` or a more relevant value.

Reply:
Right, but neither of the API requests to `/channels/:id/permissions` (except `POST`) return permission overwrites - which would be even more consistent if the API did so. And regarding to your point, neither of the managers have this behavior: `channel.permissionOverwrites.create(...).delete(..).set(..)`, so I'm kind of against this inconsistency. Although, returning `GuildChannel` reasonably makes sense and is quite relevant.